File: platforms/android-fa/app/src/main/java/com/sparrowinvest/fa/ui/clients/insurance/GapAnalysisCard.kt
package com.sparrowinvest.fa.ui.clients.insurance

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.Shield
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.dp
import com.sparrowinvest.fa.data.model.CoverageGap
import com.sparrowinvest.fa.data.model.GapAnalysisResponse
import com.sparrowinvest.fa.ui.components.GlassCard
import com.sparrowinvest.fa.ui.components.IconContainer
import com.sparrowinvest.fa.ui.theme.Primary
import com.sparrowinvest.fa.ui.theme.Success
import com.sparrowinvest.fa.ui.theme.Error
import com.sparrowinvest.fa.ui.theme.Spacing

@Composable
fun GapAnalysisCard(
    gapAnalysis: GapAnalysisResponse?,
    modifier: Modifier = Modifier
) {
    GlassCard(modifier = modifier) {
        Column(
            verticalArrangement = Arrangement.spacedBy(Spacing.compact)
        ) {
            // Header
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(Spacing.small)
            ) {
                IconContainer(
                    size = 32.dp,
                    backgroundColor = Primary.copy(alpha = 0.15f)
                ) {
                    Icon(
                        imageVector = Icons.Default.Shield,
                        contentDescription = null,
                        modifier = Modifier.size(16.dp),
                        tint = Primary
                    )
                }
                Text(
                    text = "Coverage Gap Analysis",
                    style = MaterialTheme.typography.titleSmall,
                    color = MaterialTheme.colorScheme.onSurface
                )
            }

            if (gapAnalysis != null) {
                CoverageBar(
                    label = "Life Insurance",
                    icon = Icons.Default.Shield,
                    gap = gapAnalysis.life
                )
                CoverageBar(
                    label = "Health Insurance",
                    icon = Icons.Default.Favorite,
                    gap = gapAnalysis.health
                )
            } else {
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(vertical = Spacing.medium),
                    contentAlignment = Alignment.Center
                ) {
                    CircularProgressIndicator(
                        modifier = Modifier.size(24.dp),
                        strokeWidth = 2.dp,
                        color = Primary
                    )
                }
            }
        }
    }
}

@Composable
private fun CoverageBar(
    label: String,
    icon: ImageVector,
    gap: CoverageGap
) {
    val progress = if (gap.recommended > 0) (gap.current / gap.recommended).toFloat().coerceIn(0f, 1f) else 0f
    val barColor = if (gap.adequate) Success else Error

    Column(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(8.dp))
            .background(MaterialTheme.colorScheme.surfaceVariant.copy(alpha = 0.3f))
            .padding(Spacing.compact),
        verticalArrangement = Arrangement.spacedBy(4.dp)
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Row(
                horizontalArrangement = Arrangement.spacedBy(4.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Icon(
                    imageVector = icon,
                    contentDescription = null,
                    modifier = Modifier.size(14.dp),
                    tint = barColor
                )
                Text(
                    text = label,
                    style = MaterialTheme.typography.labelMedium,
                    color = MaterialTheme.colorScheme.onSurface
                )
            }

            if (gap.adequate) {
                Row(
                    horizontalArrangement = Arrangement.spacedBy(2.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        imageVector = Icons.Default.CheckCircle,
                        contentDescription = null,
                        modifier = Modifier.size(12.dp),
                        tint = Success
                    )
                    Text(
                        text = "Adequate",
                        style = MaterialTheme.typography.labelSmall,
                        color = Success
                    )
                }
            } else {
                Text(
                    text = "Gap: ${formatAmount(gap.gap)}",
                    style = MaterialTheme.typography.labelSmall,
                    color = Error
                )
            }
        }

        // Progress bar
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(6.dp)
                .clip(RoundedCornerShape(3.dp))
                .background(MaterialTheme.colorScheme.surfaceVariant)
        ) {
            Box(
                modifier = Modifier
                    .fillMaxWidth(fraction = progress)
                    .height(6.dp)
                    .clip(RoundedCornerShape(3.dp))
                    .background(barColor)
            )
        }

        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            Text(
                text = "Current: ${formatAmount(gap.current)}",
                style = MaterialTheme.typography.labelSmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
            Text(
                text = "Recommended: ${formatAmount(gap.recommended)}",
                style = MaterialTheme.typography.labelSmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
        }
    }
}

private fun formatAmount(amount: Double): String = when {
    amount >= 10000000 -> "₹${"%.1f".format(amount / 10000000)} Cr"
    amount >= 100000 -> "₹${"%.1f".format(amount / 100000)} L"
    else -> "₹${"%,.0f".format(amount)}"
}

File: platforms/ios-fa/SparrowInvestFA/Views/Clients/Insurance/GapAnalysisCard.swift
import SwiftUI

struct GapAnalysisCard: View {
    let gapAnalysis: GapAnalysis?
    @Environment(\.colorScheme) private var colorScheme

    var body: some View {
        VStack(alignment: .leading, spacing: AppTheme.Spacing.compact) {
            HStack(spacing: AppTheme.Spacing.small) {
                ZStack {
                    RoundedRectangle(cornerRadius: AppTheme.CornerRadius.small, style: .continuous)
                        .fill(AppTheme.primary.opacity(0.1))
                        .frame(width: 28, height: 28)

                    Image(systemName: "chart.bar.doc.horizontal")
                        .font(.system(size: 12))
                        .foregroundColor(AppTheme.primary)
                }

                Text("Coverage Gap Analysis")
                    .font(AppTheme.Typography.accent(14))
                    .foregroundColor(.primary)

                Spacer()
            }

            if let gap = gapAnalysis {
                coverageBar(
                    label: "Life Insurance",
                    icon: "shield.checkered",
                    current: gap.life.current,
                    recommended: gap.life.recommended,
                    adequate: gap.life.adequate
                )

                coverageBar(
                    label: "Health Insurance",
                    icon: "heart.text.square",
                    current: gap.health.current,
                    recommended: gap.health.recommended,
                    adequate: gap.health.adequate
                )
            } else {
                HStack {
                    Spacer()
                    ProgressView()
                        .padding(.vertical, AppTheme.Spacing.medium)
                    Spacer()
                }
            }
        }
        .glassCard(cornerRadius: AppTheme.CornerRadius.large, padding: AppTheme.Spacing.compact)
    }

    private func coverageBar(label: String, icon: String, current: Double, recommended: Double, adequate: Bool) -> some View {
        let progress = recommended > 0 ? min(current / recommended, 1.0) : 0
        let barColor = adequate ? Color(hex: "10B981") : Color(hex: "EF4444")

        return VStack(alignment: .leading, spacing: 6) {
            HStack {
                Image(systemName: icon)
                    .font(.system(size: 11))
                    .foregroundColor(barColor)

                Text(label)
                    .font(AppTheme.Typography.accent(12))
                    .foregroundColor(.primary)

                Spacer()

                if adequate {
                    HStack(spacing: 2) {
                        Image(systemName: "checkmark.circle.fill")
                            .font(.system(size: 10))
                        Text("Adequate")
                            .font(AppTheme.Typography.label(10))
                    }
                    .foregroundColor(Color(hex: "10B981"))
                } else {
                    Text("Gap: \(formatAmount(recommended - current))")
                        .font(AppTheme.Typography.label(10))
                        .foregroundColor(Color(hex: "EF4444"))
                }
            }

            // Progress bar
            GeometryReader { geo in
                ZStack(alignment: .leading) {
                    RoundedRectangle(cornerRadius: 4)
                        .fill(colorScheme == .dark ? Color.white.opacity(0.08) : Color.black.opacity(0.06))
                        .frame(height: 8)

                    RoundedRectangle(cornerRadius: 4)
                        .fill(
                            LinearGradient(
                                gradient: Gradient(colors: [barColor, barColor.opacity(0.7)]),
                                startPoint: .leading, endPoint: .trailing
                            )
                        )
                        .frame(width: geo.size.width * CGFloat(progress), height: 8)
                }
            }
            .frame(height: 8)

            HStack {
                Text("Current: \(formatAmount(current))")
                    .font(AppTheme.Typography.label(10))
                    .foregroundColor(.secondary)

                Spacer()

                Text("Recommended: \(formatAmount(recommended))")
                    .font(AppTheme.Typography.label(10))
                    .foregroundColor(.secondary)
            }
        }
        .padding(AppTheme.Spacing.compact)
        .background(
            RoundedRectangle(cornerRadius: AppTheme.CornerRadius.medium, style: .continuous)
                .fill(colorScheme == .dark ? Color.white.opacity(0.04) : Color.black.opacity(0.02))
        )
    }

    private func formatAmount(_ amount: Double) -> String {
        if amount >= 10000000 {
            return String(format: "₹%.1f Cr", amount / 10000000)
        } else if amount >= 100000 {
            return String(format: "₹%.1f L", amount / 100000)
        }
        return String(format: "₹%.0f", amount)
    }
}
